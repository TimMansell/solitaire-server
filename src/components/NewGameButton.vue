<template>
  <RouterLink to="/new">
    <Button
      :is-stacked="true"
      :disabled="isGameLoading"
      data-test="new-game-btn"
    >
      New
    </Button>
  </RouterLink>
</template>

<script>
import { mapGetters } from 'vuex';
import Button from './Button.vue';

export default {
  name: 'NewGame',
  components: {
    Button,
  },
  computed: {
    ...mapGetters(['isGameLoading']),
  },
};
</script>
