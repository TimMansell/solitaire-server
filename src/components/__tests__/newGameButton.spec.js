import { shallowMount } from '@vue/test-utils';
import NewGameButton from '@/components/NewGameButton.vue';

describe('NewGameButton.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(NewGameButton, {
      computed: {
        game: () => ({
          id: 1,
        }),
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
