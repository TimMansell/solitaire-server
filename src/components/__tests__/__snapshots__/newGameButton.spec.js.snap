// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewGameButton.vue matches snapshot 1`] = `
<button-stub data-test="new-game-btn">
  New Game
</button-stub>
`;
